table/sst_file_writer.cc

```diff
@@ -237,12 +237,17 @@ Status SstFileWriter::Open(const std::string& file_path) {
     r->column_family_name = "";
     cf_id = TablePropertiesCollectorFactory::Context::kUnknownColumnFamily;
   }
-
+  std::string db_session_id = r->ioptions.env->GenerateUniqueId();
```

Comment:
I'm not very clear here. Assume I'm an application C, C opens a DB and the DB get a session ID. After some time, C calls SstFileWriter and dump out an SST file from the DB. Based on this line `std::string db_session_id = r->ioptions.env->GenerateUniqueId();`, the generated SST file's db_session_id will be different from the DB that C opens?

Reply:
Yes. I think there is no way for SstFileWriter to reference back any DB because SstFileWriter is used to create SST files to be added to some DB later. (Please correct me if I'm misunderstood something :)

So here, in this way, we ensure that every time one opens SstFileWriter, it behaves like one opens a "DB".

Reply:
It would be better to explain this special case in the comment for user to understand.